### ISSUE TITLE
Bad predictions with RetinaNet 

### ISSUE THREAD

**Issue by hugopi** [keras-team/keras-cv#1828]
Dear KerasCV TEAM,

I hope this message finds you well. Firstly, I would like to express my sincere appreciation for the incredible work you and your team have done in developing the kerasCV library. It has been an invaluable asset in my projects, and I am grateful for the effort you've put into creating such a powerful tool.

However, recently, I have been encountering some challenges related to the predictions generated by the RetinaNet model. Despite following the recommended guidelines and ensuring the dataset's quality, I have noticed that predictions are really poor even if my model is converging.

Given your expertise in this domain, I was wondering if you could shed some light on this matter. I will give you some of my results, inputs and configuration.

I understand that debugging such complex models can be challenging, and I want to assure you that I deeply value the hard work and dedication that went into developing this library. I am optimistic that with your guidance and expertise, we can address these prediction issues and leverage the full potential of the RetinaNet model.

Thank you once again for your outstanding contributions to the Computer Vision community. I eagerly await your response and look forward to continuing to work with this remarkable tool.

Best regards,

Hugo PIRES

Inputs : 
```
>>>train_dataset.element_spec
(TensorSpec(shape=(8, 512, 512, 3), dtype=tf.float32, name=None), {'boxes': RaggedTensorSpec(TensorShape([8, None, 4]), tf.float32, 1, tf.int64), 'classes': RaggedTensorSpec(TensorShape([8, None]), tf.float32, 1, tf.int64)})
```
```
>>>next(iter(train_dataset))

(<tf.Tensor: shape=(8, 512, 512, 3), dtype=float32, numpy=
array([[[[0.22794858, 0.02173912, 0.22794858],
         [0.13921069, 0.02173912, 0.13921069],
         [0.07489516, 0.02173912, 0.07489516],
         ...,
         [0.12426693, 0.02173912, 0.12426693],
         [0.13279389, 0.02173912, 0.13279389],
         [0.15347545, 0.02173912, 0.15347545]],
        [[0.23022266, 0.02173912, 0.23022266],
         [0.15261193, 0.02173912, 0.15261193],
         [0.09595545, 0.02173912, 0.09595545],
         ...,
         [0.09794039, 0.02173912, 0.09794039],
         [0.17421728, 0.02173912, 0.17421728],
         [0.24660785, 0.02173912, 0.24660785]],
        [[0.22908357, 0.02173912, 0.22908357],
         [0.16956106, 0.02173912, 0.16956106],
         [0.12426693, 0.02173912, 0.12426693],
         ...,
         [0.08325624, 0.02173912, 0.08325624],
         [0.1937029 , 0.02173912, 0.1937029 ],
         [0.2968543 , 0.02173912, 0.2968543 ]],
        ...,
        [[0.02680415, 0.02173912, 0.02680415],
         [0.02437472, 0.02173912, 0.02437472],
         [0.03012604, 0.02173912, 0.03012604],
         ...,
        [...]
         ...,
         [0.9782608 , 0.02173912, 0.9782608 ],
         [0.9782608 , 0.02173912, 0.9782608 ],
         [0.9782608 , 0.02173912, 0.9782608 ]],
        [[0.88584316, 0.02173912, 0.88584316],
         [0.6117155 , 0.02173912, 0.6117155 ],
         [0.6360084 , 0.02173912, 0.6360084 ],
         ...,
         [0.9782608 , 0.02173912, 0.9782608 ],
         [0.78946066, 0.02173912, 0.78946066],
         [0.6478084 , 0.02173912, 0.6478084 ]],
        ...,
        [[0.6295348 , 0.02173912, 0.6295348 ],
         [0.72861284, 0.02173912, 0.72861284],
         [0.7831771 , 0.02173912, 0.7831771 ],
         ...,
         [0.5030216 , 0.02173912, 0.5030216 ],
         [0.559709  , 0.02173912, 0.559709  ],
         [0.6218429 , 0.02173912, 0.6218429 ]],
        [[0.8464153 , 0.02173912, 0.8464153 ],
         [0.9509264 , 0.02173912, 0.9509264 ],
         [0.69837785, 0.02173912, 0.69837785],
         ...,
         [0.5980263 , 0.02173912, 0.5980263 ],
         [0.5246342 , 0.02173912, 0.5246342 ],
         [0.5213425 , 0.02173912, 0.5213425 ]],
        [[0.9782608 , 0.02173912, 0.9782608 ],
         [0.9782608 , 0.02173912, 0.9782608 ],
         [0.6231193 , 0.02173912, 0.6231193 ],
         ...,
         [0.83307904, 0.02173912, 0.83307904],
         [0.5690824 , 0.02173912, 0.5690824 ],
         [0.46206063, 0.02173912, 0.46206063]]]], dtype=float32)>, 
         {'boxes': <tf.RaggedTensor [[[484.0, 293.0, 12.0, 29.0]], 
                                                      [[359.0, 76.0, 10.0, 13.0]],
                                                      [[197.0, 238.0, 10.0, 10.0]], 
                                                      [[43.0, 91.0, 10.0, 10.0],[349.0, 323.0, 10.0, 10.0]],
                                                      [[276.0, 345.0, 8.0, 10.0]], [[474.0, 82.0, 10.0, 10.0]],
                                                      [[137.0, 202.0, 6.0, 6.0],[137.0, 192.0, 2.0, 6.0],[468.0, 143.0, 12.0, 12.0]], 
                                                      [[281.0, 188.0, 8.0, 12.0],[211.0, 359.0, 8.0, 14.0],[324.0, 340.0, 22.0, 20.0]]]>, 
          'classes': <tf.RaggedTensor [[1.0], [0.0], [1.0], [1.0, 1.0], [1.0], [1.0], [1.0, 1.0, 1.0],[1.0, 1.0, 1.0]]>})
          
```
Config : 
https://gist.github.com/hugopi/c98951f81bb4b918d5b5d9ad8209a33b

Tensor board logs : 
box loss
![image](https://github.com/keras-team/keras-cv/assets/90212732/57a77b21-bb40-416c-9e91-f647a56302a1)
class loss
![image](https://github.com/keras-team/keras-cv/assets/90212732/49784d06-85d0-4b32-a45e-8111707b1d2d)
epoch_percent_boxes_matched_with_anchor
![image](https://github.com/keras-team/keras-cv/assets/90212732/1a102ffb-5e7a-449d-b395-ee6db6e2a08e)

Results : 
0.0 = no boat 
1.0 = boat
Green box = predictions 
Red box = ground truth 
test1
![image](https://github.com/keras-team/keras-cv/assets/90212732/faacd803-e734-462c-88f1-1b2ffd896d2a)
test2.1
![image](https://github.com/keras-team/keras-cv/assets/90212732/8b36ad76-cdbb-4b6a-baa0-9c1e4a08855a)
test2.2
![image](https://github.com/keras-team/keras-cv/assets/90212732/c589ef19-3fc6-4f97-8e4f-73f8b158f025)
test3
![image](https://github.com/keras-team/keras-cv/assets/90212732/6e94bbf8-07be-4403-bf04-3206e42c9924)
test4
![image](https://github.com/keras-team/keras-cv/assets/90212732/ed602889-fe49-491c-9c97-99ba04f1fe5e)

**Comment by jbischof** [comment]
@hugopi it's hard know with certainty but those learning curves don't seem particularly healthy. They are basically flat with huge drops. Have you tried lowering the learning rate to get a more consistent (and likely generalizable) improvement to loss?

**Comment by hugopi** [comment]
The huge drops are due to the backbone tuning. In this training I did not use early stopping so the model train until the last training epoch and then start training with backbone trainable= True. 

Concerning learning rate I use ReduceOnPlateau : 

 reduce_on_plateau = tf.keras.callbacks.ReduceLROnPlateau(
        monitor="val_loss",
        factor=0.2,
        patience=5,
        min_lr=0.00000000001
    )
    
  Is it a good practice ?

**Comment by jbischof** [comment]
You should make the backbone tunable from the start and either use a very low learning rate for debugging or a learning rate schedule as we discuss in the OD guide.

Your problem is actually very easy: detect something not black! There should be no issue getting a good fit

**Comment by hugopi** [comment]
HI @LukeWood , 

This dataset comes from Xview competition Dark vessels. (Vessel detection on Radar images) 

This a link to the paper and the dataset : 

https://iuu.xview.us/xview3_dataset_whitepaper.pdf
https://iuu.xview.us/dataset
